app/src/Pages/Portal/Contributions/Utils/ContributionsFields.js

```diff
@@ -226,167 +230,175 @@ export const fields = {
-        label: 'Fundraising Event Expenses',
-      },
-      {
-        value: 'general_operating_expenses',
-        label: 'General Operating Expenses',
-      },
-      { value: 'printing', label: 'Literature/Brochures/Printing' },
-      { value: 'management', label: 'Management Services' },
-      {
-        value: 'print_advertising',
-        label: 'Newspaper and Other Periodical Advertising',
-      },
-      { value: 'other_advertising', label: 'Other Advertising' },
-      { value: 'petition_Circulators', label: 'Petition Circulators' },
-      { value: 'postage', label: 'Postage' },
-      {
-        value: 'preparation_of_advertising',
-        label: 'Preparation and Production of Advertising',
-      },
-      { value: 'surveys_and_polls', label: 'Surveys and Polls' },
-      { value: 'travel_expenses', label: 'Travel Expenses' },
-      { value: 'utilities', label: 'Utilities' },
-      { value: 'wages', label: 'Wages/Salaries/Benefits' },
-    ]
-  ),
+  inKindDescription: {
+    label: 'Inkind Description',
+    section: FormSectionEnum.OTHER_DETAILS,
+    component: TextField,
+    validation: Yup.string(),
+  },
+  inKindType: {
+    // TODO Check that is posts all the way to the DB
+    label: 'Inkind Type',
+    section: FormSectionEnum.OTHER_DETAILS,
+    component: SelectField,
+    validation: Yup.string(),
+    options: {
+      values: [
+        { value: 'broadcast_advertising', label: 'Broadcast Advertising' },
+        {
+          value: 'fundraising_event_expenses',
+          label: 'Fundraising Event Expenses',
+        },
+        {
+          value: 'general_operating_expenses',
+          label: 'General Operating Expenses',
+        },
+        { value: 'printing', label: 'Literature/Brochures/Printing' },
+        { value: 'management', label: 'Management Services' },
+        {
+          value: 'print_advertising',
+          label: 'Newspaper and Other Periodical Advertising',
+        },
+        { value: 'other_advertising', label: 'Other Advertising' },
+        { value: 'petition_Circulators', label: 'Petition Circulators' },
+        { value: 'postage', label: 'Postage' },
+        {
+          value: 'preparation_of_advertising',
+          label: 'Preparation and Production of Advertising',
+        },
+        { value: 'surveys_and_polls', label: 'Surveys and Polls' },
+        { value: 'travel_expenses', label: 'Travel Expenses' },
+        { value: 'utilities', label: 'Utilities' },
+        { value: 'wages', label: 'Wages/Salaries/Benefits' },
+      ],
+    },
+  },
   // Not required if occupation & employer name/address filled in
-  occupationLetterDate: formField(
-    'Occupation Letter Date',
-    FormSectionEnum.OTHER_DETAILS,
-    DateField,
-    Yup.date('Enter occupation letter date')
-  ),
+  occupationLetterDate: {
+    label: 'Occupation Letter Date',
+    section: FormSectionEnum.OTHER_DETAILS,
+    component: DateField,
+    validation: Yup.date(),
```

Comment:
Are these warnings being implemented in some other way, or removed?

Reply:
Still needed. Yup validates static conditions like type and always required. Validate is conditional like values visible if and required if.

Reply:
Yup is used for static checks like type and validate for dynamic checks

Reply:
Yes to still relevant.

Reply:
Renamed to extractValidateErrors,
Yup should have errors attached for static uncontrolled fields. Otherwise it's handled by validate and in the date field case and others the input is controlled by a component or mask.